src/main/java/eu/openanalytics/containerproxy/backend/ecs/EcsBackend.java

```diff
@@ -0,0 +1,346 @@
+import eu.openanalytics.containerproxy.model.spec.ContainerSpec;
+import eu.openanalytics.containerproxy.model.spec.ProxySpec;
+import eu.openanalytics.containerproxy.util.Retrying;
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.ContainerPortBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import org.springframework.data.util.Pair;
+import org.springframework.security.core.Authentication;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ecs.EcsClient;
+import software.amazon.awssdk.services.ecs.model.*;
+
+public class EcsBackend extends AbstractContainerBackend {
+
+    private static final String PROPERTY_PREFIX = "proxy.ecs.";
+    private static final String PROPERTY_CLUSTER = "name";
+    private static final String PROPERTY_REGION = "region";
+
+    private EcsClient ecsClient;
+
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        Region region = Region.of(getProperty(PROPERTY_REGION));
+
+        ecsClient = EcsClient.builder()
+                .region(region)
+                .build();
+    }
+
+    public void initialize(EcsClient client) {
+        super.initialize();
+        ecsClient = client;
+    }
+
+
+    private Map<RuntimeValueKey<?>, RuntimeValue> parseTagsAsRuntimeValues(String taskArn, String containerId) {
+        Map<RuntimeValueKey<?>, RuntimeValue> runtimeValues = new HashMap<>();
+
+        HashMap<String, String> tags = ecsClient.listTagsForResource(builder -> builder.resourceArn(taskArn)).tags().stream().collect(
+                HashMap::new,
+                (m, v) -> m.put(v.key(), v.value()),
+                HashMap::putAll
+        );
+
+
+        for (RuntimeValueKey<?> key : RuntimeValueKeyRegistry.getRuntimeValueKeys()) {
+            if (key.getIncludeAsTag()) {
+                String value = tags.get(key.getKeyAsTag());
+                if (value != null) {
+                    runtimeValues.put(key, new RuntimeValue(key, key.deserializeFromString(value)));
+                } else if (key.isRequired()) {
+                    // value is null but is required
+                    log.warn("Ignoring container {} because no tag named {} is found!", containerId, key.getKeyAsLabel());
+                    return null;
+                }
+            }
+        }
+
+        return runtimeValues;
+    }
+
+    @Override
+    public List<ExistingContainerInfo> scanExistingContainers() throws Exception {
+        // TODO Auto-generated method stub
+
+        ArrayList<ExistingContainerInfo> containers = new ArrayList<>();
+
+        ListTasksResponse ecsTasks = ecsClient.listTasks(builder -> builder.cluster(getProperty(PROPERTY_CLUSTER)));
+        List<String> taskArns = ecsTasks.taskArns();
+
+        DescribeTasksResponse tasks = ecsClient
+                .describeTasks(builder -> builder.cluster(getProperty(PROPERTY_CLUSTER)).tasks(taskArns));
```

Comment:
Really unsure if I am reading this one correctly, but will this implementation handle the case where `taskArns` contains more than 100 items? https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/ecs/model/DescribeTasksRequest.Builder.html